googletest/cmake/internal_utils.cmake

```diff
@@ -196,6 +187,19 @@ function(cxx_library_with_type name type cxx_flags)
   endif()
 
   target_compile_features(${name} PUBLIC cxx_std_17)
+
+  if (NOT BUILD_SHARED_LIBS AND NOT gtest_force_shared_crt)
```

Comment:
Can `AND NOT DEFINED CMAKE_MSVC_RUNTIME_LIBRARY` be added here so that it can be controlled from without using the property's initializer variable?

Reply:
How about deferring to `CMAKE_MSVC_RUNTIME_LIBRARY` if it is set and deprecating `gtest_force_shared_crt`'s use in conjunction with it?

Something like:

```cmake
if (NOT DEFINED CMAKE_MSVC_RUNTIME_LIBRARY)
  option(gtest_force_shared_crt …)
  if (some_suitable_condition)
    message(DEPRECATION
      "gtest_force_shared_crt is deprecated in preference to `CMAKE_MSVC_RUNTIME_LIBRARY` instead")
  endif ()
  if (gtest_force_shared_crt)
    set(CMAKE_MSVC_RUNTIME_LIBRARY "…")
  else ()
    set(CMAKE_MSVC_RUNTIME_LIBRARY "…")
  endif ()
elseif (DEFINED gtest_force_shared_crt)
  message(DEPRECATION
    "Since CMAKE_MSVC_RUNTIME_LIBRARY is set, gtest_force_shared_crt settings are ignored")
endif ()
```

Reply:
You can do something like:

```cmake
MultiThreaded$<$<CONFIG:Debug>:Debug>$<$<OR:$<BOOL:${gtest_force_shared_crt}>,$<BOOL:${BUILD_SHARED_LIBS}>>:DLL>
```

Reply:
I have reverted the change that would deprecate gtest_force_shared_crt and simply add the condition to not set CMAKE_MSVC_RUNTIME_LIBRARY if it is already defined, as you asked.

Would rather this be a "No rock the boat" PR so that it is easy for maintainers to accept.

Edit: Also I reverted the changes to the readme since they were only needed with deprecation.